src/plugins/csp_configuration_provider/README.md

```diff
@@ -0,0 +1,64 @@
+# CspConfigurationProvider
+
+A OpenSearch Dashboards plugin
+
+This plugin introduces the support of self service dynamic configuration of Content Security Policy (CSP) rules without requiring a server restart. It registers a pre-response handler to `HttpServiceSetup` which can get CSP rules from a new index `.opensearch_dashboards_config` and then rewrite to CSP header. OSD users could update the content of the index with new CSP rules and the change will take effect immediately.
```

Comment:
on start up I believe we can give the option to populate this index and give the mapping.

One thing I'm worried about is the whole `.` usage in OSD not being actually treated as a system index and any issues if OpenSearch does any changes to enforce to pre-pending `.`

Reply:
1. I intentionally do not create the new index to avoid race condition. (The existing `.kibana` often runs into this problem when there are multiple nodes). Furthermore, for customers who do not want to change the default behavior, there is no need for the new index to exist. 

Currently there is only one doc in the index with only one field. It looks like that an index mapping may not be necessary yet. Furthermore, it may be infeasible to define a mapping for a non-exist index.

2. I read through that issue about system index and notice this comment https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4713#issuecomment-1835865775 The way we name the new index will make it match the current regular expression and get the benefits provided by OpenSearch core.